src/Module/Api/Twitter/DirectMessages/Destroy.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2021, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Api\Twitter\DirectMessages;
+
+use Friendica\Database\DBA;
+use Friendica\Module\BaseApi;
+use Friendica\Network\HTTPException\BadRequestException;
+
+/**
+ * delete a direct_message from mail table through api
+ *
+ * @see   https://developer.twitter.com/en/docs/direct-messages/sending-and-receiving/api-reference/delete-message
+ */
+class Destroy extends BaseApi
+{
+	protected function rawContent(array $request = [])
+	{
+		BaseApi::checkAllowedScope(BaseApi::SCOPE_WRITE);
+		$uid = BaseApi::getCurrentUserID();
+
+		$id      = filter_var($request['id'] ?? 0,                    FILTER_VALIDATE_INT);
+		$verbose = filter_var($request['friendica_verbose'] ?? false, FILTER_VALIDATE_BOOLEAN);
+
+		$parenturi = $request['friendica_parenturi'] ?? '';
+
+		// error if no id or parenturi specified (for clients posting parent-uri as well)
+		if ($verbose && ($id == 0 || $parenturi == "")) {
+			$answer = ['result' => 'error', 'message' => 'message id or parenturi not specified'];
+			$this->response->exit('direct_messages_delete', ['direct_messages_delete' => $answer], $this->parameters['extension'] ?? null);
+			return;
+		}
+
+		// BadRequestException if no id specified (for clients using Twitter API)
+		if ($id == 0) {
+			throw new BadRequestException('Message id not specified');
+		}
+
+		// add parent-uri to sql command if specified by calling app
+		$sql_extra = ($parenturi != "" ? " AND `parent-uri` = '" . DBA::escape($parenturi) . "'" : "");
+
+		// error message if specified id is not in database
+		if (!DBA::exists('mail', ["`uid` = ? AND `id` = ? " . $sql_extra, $uid, $id])) {
```

Comment:
```suggestion
		if (!$this->dba->exists('mail', ["`uid` = ? AND `id` = ? " . $sql_extra, $uid, $id])) {
```
And add `protected $dba;` as property